### PR TITLE
improved template

### PR DESCRIPTION
**Pull request by Mouvedia**
check the comments on the code for details

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/LeeCheneler/umd-generator/pull/1" target="_blank">ReviewNinja

File: index.js [hunk 1]
```diff
@@ -16,24 +16,27 @@
...
-        root.<MODULE_NAME> = factory(/* dependencies */);
-    }
-}(this, function(/* dependencies */) {
-    var <MODULE_NAME> = {
-        // MODULE CODE HERE....
-    }
-    return <MODULE_NAME>;
+  if (typeof define === 'function' && define.amd)
+    define([/* dependencies */], factory);
+  else if (typeof exports === 'object' && !!exports && !exports.nodeType) {
```
Comment (Mouvedia) [inline index.js]: - is not `null`
- https://www.w3.org/Bugs/Public/show_bug.cgi?id=11960

File: index.js [hunk 2]
```diff
@@ -16,24 +16,27 @@
...
+    if (typeof module === 'object' && !!module && module.exports)
+      module.exports = factory(/* dependencies */);
+    else
+      exports['default'] = factory(/* dependencies */);
```
Comment (Mouvedia) [inline index.js]: method name chosen to match the ES6 export statement counterpart
discuss: 3rd parameter

File: index.js [hunk 3]
```diff
@@ -16,24 +16,27 @@
...
+  } else if (typeof YUI === 'function' && YUI.add)
```
Comment (Mouvedia) [inline index.js]: YUI support

File: index.js [hunk 4]
```diff
@@ -16,24 +16,27 @@
...
+    YUI.add('<MODULE_NAME>', function (Y) { Y['default'] = factory(); });
+  else
+    root['<MODULE_NAME>'] = factory();
```
Comment (Mouvedia) [inline index.js]: `moduleName` might contain a string

File: index.js [hunk 5]
```diff
@@ -16,24 +16,27 @@
...
+}((function () {
+  try {
+    return Function('return this')() || (42, eval)('this');
+  } catch (e) {
+    return self;
```
Comment (Mouvedia) [inline index.js]: fallback for script-src and default-src directives

File: index.js [hunk 6]
```diff
@@ -16,24 +16,27 @@
...
-    }
-}(this, function(/* dependencies */) {
-    var <MODULE_NAME> = {
-        // MODULE CODE HERE....
-    }
-    return <MODULE_NAME>;
+  if (typeof define === 'function' && define.amd)
+    define([/* dependencies */], factory);
+  else if (typeof exports === 'object' && !!exports && !exports.nodeType) {
+    if (typeof module === 'object' && !!module && module.exports)
```
Comment (Mouvedia) [inline index.js]: Node.js compatible

File: index.js [hunk 7]
```diff
@@ -16,24 +16,27 @@
...
+      module.exports = factory(/* dependencies */);
+    else
```
Comment (Mouvedia) [inline index.js]: CJS fallback

**Comment by LeeCheneler** [comment]
LGTM

:+1:

**Pull request merged**